cosmos/models/Workflow.py

```diff
@@ -443,6 +446,17 @@ def delete(self, delete_files=False):
         self.session.commit()
         print >> sys.stderr, '%s Deleted' % self
 
+    def get_first_failed_task(self, key=lambda t: t.finished_on):
+        """
+        Return the first failed Task (chronologically).
+
+        If no Task failed, return None.
+        """
+        for t in sorted(self.task_graph(), key=key):
```

Comment:
```
>>> sorted([datetime.datetime.now(), None])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't compare datetime.datetime to NoneType
```

Apart from the fact that this doesn't work, I think it's pretty good! let me know what you think ;-)